resources-library/src/jsSharedMain/kotlin/Resource.kt

```diff
@@ -80,31 +92,17 @@ public actual class Resource actual constructor(path: String) {
      * Node-based resource implementation.
      */
     private class ResourceNode(val path: String) {
-        val fs = nodeRequire("fs")
-
-        private fun nodeRequire(name: String): dynamic {
-            // Alternative to declaring `private external fun require(name: String): dynamic` and
-            // using `require("fs")` directly, since it will cause webpack to complain when running
-            // on the browser, even though the code is unused.
-
-            return try {
-                js("module['' + 'require']")(name)
-            } catch (e: dynamic) {
-                throw IllegalArgumentException("Module not found: $name", e as? Throwable)
-            }
-        }
-
-        fun exists(): Boolean = fs.existsSync(path) as Boolean
+        fun exists(): Boolean = existsSync(path)
 
         fun readText(): String = runCatching {
-            fs.readFileSync(path, "utf8") as String
+            readFileSync(path, "utf8")
         }.getOrElse { cause ->
             throw FileReadException("$path: Read failed", cause)
         }
 
         fun readBytes(): ByteArray = runCatching {
-            val buffer = fs.readFileSync(path).unsafeCast<Uint8Array>()
-            Int8Array(buffer.buffer, buffer.byteOffset, buffer.length).unsafeCast<ByteArray>()
+            val buffer = readFileSync(path)
+            Int8Array(buffer.buffer, buffer.byteOffset, buffer.length) as ByteArray
```

Comment:
There must be a better way to do this.

Reply:
@goncalossilva is this because `unsafeCast` is available only under the `js*` source sets?

Reply:
`unsafeCast` is available on Wasm too, but its type must inherit from `JsAny` which `ByteArray` doesn't.

Reply:
@goncalossilva maybe you can introduce your own `unsafeCast` temporarely? It's just a call to `asDynamic`.

Reply:
Ah just seen, it's `implementedAsIntrinsic`, so no luck on that.

Reply:
I've just added support for `wasmJs` to `antlr-kotlin` and I've experimented a bit with it.

Working with primitive types in a shared way, with JS interop like this case, doesn't seem to be possible.  
I ended up duplicating some of the code and using WASM types like `JsBoolean` and `JsNumber`, that you can get with, e.g.:

```
yourBool.toJsBoolean()
```

Reply:
If you need WASI file system APIs, I've been pointed to https://github.com/kowasm/kowasm  
You'll need to copy-paste the WASM imports and Kotlin functions, but they should work.

Reply:
@lppedd nice share, thanks! I have some downtime during the holidays, but I'll definitely be looking into these when I pick this back up.